sql/core/src/test/scala/org/apache/spark/sql/scripting/SqlScriptingInterpreterSuite.scala

```diff
@@ -362,4 +367,48 @@ class SqlScriptingInterpreterSuite extends QueryTest with SharedSparkSession {
       verifySqlScriptResult(commands, expected)
     }
   }
+
+  test("if's condition must be a boolean statement") {
+    withTable("t") {
+      val commands =
+        """
+          |BEGIN
+          |  IF 1 THEN
+          |    SELECT 45;
+          |  END IF;
+          |END
+          |""".stripMargin
+      checkError(
+        exception = intercept[SqlScriptingException] (
+          runSqlScript(commands)
+        ),
+        errorClass = "INVALID_BOOLEAN_STATEMENT",
+        parameters = Map("invalidStatement" -> "1")
+      )
+    }
+  }
+
+  test("if's condition must return a single row data") {
```

Comment:
@davidm-db 
Although I have split it into `2 error conditions` (`INVALID_BOOLEAN_STATEMENT` and `BOOLEAN_STATEMENT_TOO_MANY_ROWS`) as per your suggestion, but unfortunately, it seems impossible to hit it, as shown in this case.
It hit an error `SCALAR_SUBQUERY_TOO_MANY_ROWS`.

Reply:
Nice catch! I'm honestly not sure if there is a way to hit our check then, but also I'm not sure there isn't... So, I would leave the check just in case, but do the tests cases as you did them and that should keep us covered.

@cloud-fan what do you think? Do you know if `ScalarSubquery` is always created in boolean expressions? Should we rely on that, or do as I said before and keep the check for multiple rows just in case?